src/timething/text.py

```diff
@@ -41,12 +45,56 @@ def nums2words(text: str, lang: str):
     pronounced differently.
     """
 
-    def fn(match, to='cardinal'):
+    def fn(match, to="cardinal"):
         number = float(match.group(0))
         if number.is_integer() and number > 1800 and number < 2100:
-            to = 'year'
+            to = "year"
             number = int(number)
 
         return num2words(number, lang=lang, to=to)
 
     return re.sub(NUMS_RE, fn, text)
+
+
+# decoding
+
+
+def ctc_collapse(tokens: typing.List[str], blank="<b>", delimiter="|"):
+    "Collapse CTC blanks"
+    transcript = "".join(c for c, _ in itertools.groupby(tokens))
+    return " ".join(transcript.replace(blank, "").split(delimiter))
+
+
+def decode_best(logprobs, dictionary):
+    "Argmax decoding of P(char | audio)."
+    x = torch.argmax(logprobs, dim=-1)
+    return [dictionary[code.item()] for code in x.squeeze()]
+
+
+def decode_best_ctc(logprobs, dictionary, delimiter="|"):
+    "Argmax decoding of P(char | audio). Inclues CTC collapsing"
+
+    tokens = decode_best(logprobs, dictionary)
+    return ctc_collapse(tokens, dictionary[0], delimiter)
```

Comment:
Is `dicitionary[0]` always the blank char?

Reply:
good call – probably yes, but I moved this config to somewhere more obvious, and replaced the other defaults with '<pad>', which is what they call it in wav2vec2. ty